app/models/payment/braintree/transaction.rb

```diff
@@ -64,4 +73,15 @@ def increment_funding_counter
 
     FundingCounter.update(page: page, currency: currency, amount: amount)
   end
+
+  def set_refund_state
+    self.refund_synced = refund_was
+    true
+  end
+
+  def decrement_funding_counter
+    return true if refund_synced
+
+    FundingCounter.reduce(page: page, currency: currency, amount: amount)
+  end
```

Comment:
Instead of having an increment and decrement pair of methods, is it possible to get the difference from amount - refund or something along those lines? That way we can simply pass it to `update` without any modifications needed (incrementing a negative will decrement).